NASS-988: Tamanu Desktop login page update

mlie788 opened the pull request:
### Changes

Mostly styling and rearranging components to update the design of the login page to the new design. Added new logo and image files to the repo. 

Currently the reset code and password fields are not separated and new workflow will be done in a separate ticket.

Old login & forgot password screens:
![Screenshot 2023-12-11 at 1 50 48 PM](https://github.com/beyondessential/tamanu/assets/100409601/d13c4e2e-1eac-4916-b968-154454699a0d)
![Screenshot 2023-12-11 at 1 53 53 PM](https://github.com/beyondessential/tamanu/assets/100409601/f48dd81b-e703-4d8c-8579-3090842ed54d)
![Screenshot 2023-12-11 at 1 54 12 PM](https://github.com/beyondessential/tamanu/assets/100409601/79533b26-ebdf-4007-a160-c2654839ca5c)
<img width="1183" alt="Screenshot 2023-12-12 at 3 31 26 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/fc7a79e1-0b24-4425-948d-1f89c1394800">
<img width="1183" alt="Screenshot 2023-12-12 at 3 32 07 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/753379b8-4201-468b-bfbb-dfb601a98a9d">
<img width="1182" alt="Screenshot 2023-12-12 at 3 32 43 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/60576929-9860-4590-bf15-e32e43ecd6b2">
<img width="1183" alt="Screenshot 2023-12-12 at 3 33 49 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/27c86c09-c93a-46aa-ab49-2eb3aa6db3bd">



New login & forgot password screens:
<img width="1184" alt="Screenshot 2023-12-12 at 3 12 41 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/fce3a00d-e6e5-4aff-9b00-57ac5aff2071">
<img width="1185" alt="Screenshot 2023-12-12 at 3 12 57 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/a7d0bc38-1599-4332-8ff2-086bf51bb819">
<img width="1186" alt="Screenshot 2023-12-12 at 3 13 25 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/6af2a041-02cd-4e8b-bb72-75ca8080c32f">
<img width="1186" alt="Screenshot 2023-12-12 at 3 13 43 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/1a52882f-17bb-4b79-be13-1117d5002098">
<img width="1183" alt="Screenshot 2023-12-12 at 3 16 01 PM" src="https://github.com/beyondessential/tamanu/assets/100409601/59087ab3-3910-47df-8b40-c36fa4d3abd2">



### Checklist

- [ ] Code is finished
- [ ] Tests are written
- [ ] UI Screenshots added to the Linear issue
- [ ] Testing notes added to the Linear issue

_Upon merging:_

- [ ] Update the changelog (find it [here](https://github.com/beyondessential/tamanu/releases) in the appropriate _draft_ release)
  - [ ] with a ticket reference (e.g. `TAN-123: a one line description`, keep the lists sorted)
  - [ ] any config/settings changes and manual deployment steps
  - [ ] any db schema or other changes that could impact downstream data analysis
- [ ] Update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly) or [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q) as needed
- [ ] Update the [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c) as needed

dannash100 committed:
major wip

dannash100 committed:
Working somewhat

dannash100 committed:
Working somewhat

dannash100 committed:
fix svg and eslint

dannash100 committed:
little bit of cleanup

dannash100 committed:
build to web update and make so we can build upon it incrementally

passcod committed:
Move back to app

passcod committed:
Move back resources

passcod committed:
Disable codeship

passcod committed:
Revert yarn.lock for now

passcod committed:
Revert packages/shared-src/src/reports/vaccine-list.js

passcod committed:
Revert certificate (PDF work)

passcod committed:
Unmerge some meta files

passcod committed:
Revert images

passcod committed:
Just go away discovery

passcod committed:
Eliminate electron

passcod committed:
Last few reverts/comments

passcod committed:
Merge branch 'main' into oldanew/tav-676

passcod committed:
Merge branch 'main' into feature/tav-676-browser-edition

passcod committed:
Update yarn lock

passcod committed:
Deploy from tamanu-browser ops branch

passcod committed:
wat

passcod committed:
Merge branch 'main' into feature/tav-676-browser-edition

passcod committed:
Merge branch 'main' into feature/tav-676-browser-edition

passcod committed:
Workaround bug in Yarn

https://github.com/yarnpkg/yarn/issues/7734#issuecomment-671729912

passcod committed:
Start on vite

passcod committed:
Rename all JSX files to the jsx extension

passcod committed:
In shared

passcod committed:
In servers

passcod committed:
Merge branch 'fix/jsx-file-extensions' into feature/tav-676-browser-edition

passcod committed:
Vite it up

passcod committed:
Build shareds as es6

passcod committed:
and esnext

passcod committed:
TAV-856: Delete discovery

passcod committed:
TAV-714: no more need for /upgrade route

passcod committed:
TAV-714: don't have a desktop to bundle with servers

passcod committed:
Merge branch 'main' into feature/tav-676-browser-edition

passcod committed:
No more webpack

passcod committed:
Adjust rollup options

passcod committed:
Yarn

passcod committed:
Fix PDF worker

passcod committed:
Fix bug in invoicing with mispelled constant

passcod committed:
Split dateTime utils to not require config

passcod committed:
Remove webpack hack

passcod committed:
TAV-885: Remove server selection field

passcod committed:
TAV-885: Derive host from URL

passcod committed:
Add systemd service files

passcod committed:
Serve /api with vite in dev

passcod committed:
Fix error messages from network errors

passcod committed:
Throw if code >= 502

passcod committed:
Nope, don't do it here

passcod committed:
Update scripts

passcod committed:
Allow passing a target to build-all

passcod committed:
Compile everything in ES6 form

passcod committed:
Remove webpack from meta-server

passcod committed:
Completely remove webpack and babel

(except from mobile)

passcod committed:
Remove codeship and obsolete scripts

passcod committed:
Merge branch 'main' into feature/tav-676-browser-edition

passcod committed:
Rename all files to mjs

passcod committed:
Remove babel from eslintrc

passcod committed:
Manual chunking

passcod committed:
Revert "Rename all files to mjs"

This reverts commit a409e0271afb8a5d7d34c93a4abb920e4fd694c1.

passcod committed:
Type: module, yes, but surely it can work without mass rewrites

flynncs committed:
NASS-1041: Fix 'New form' button on web (#5025)

mlie788 committed:
Aligned signin form to the left and added heading and subtext

mlie788 committed:
Aligned signin form to the left and added heading and subtext

mlie788 committed:
Aligned remember me check box to the right

mlie788 committed:
Aligned remember me check box to the right

mlie788 committed:
Added new background image for login screen

mlie788 committed:
Added new background image for login screen

mlie788 committed:
Styled checkbox and log in form fields

mlie788 committed:
Styled checkbox and log in form fields

mlie788 committed:
Added new tamanu logo for log in

mlie788 committed:
Added new tamanu logo for log in

mlie788 committed:
Made changes to the login and forgot password button

mlie788 committed:
Made changes to the login and forgot password button

mlie788 committed:
Added version number and adjusted padding between buttons

mlie788 committed:
Added version number and adjusted padding between buttons

mlie788 committed:
Changes to background and login display to handle window resizing

mlie788 committed:
Changes to background and login display to handle window resizing

mlie788 committed:
Added page refresh functionality when tamanu logo is clicked

mlie788 committed:
Added page refresh functionality when tamanu logo is clicked

mlie788 committed:
Adjusted positioning of splash image and restyled forgot password button to text button

mlie788 committed:
Adjusted positioning of splash image and restyled forgot password button to text button

mlie788 committed:
Extra padding for log in button

mlie788 committed:
Extra padding for log in button

mlie788 committed:
Renamed Grid to Container

mlie788 committed:
Renamed Grid to Container

mlie788 committed:
Implemented ui/ux chages to the forgot password workflow

mlie788 committed:
Implemented ui/ux chages to the forgot password workflow

mlie788 committed:
Removed unused parameters and imports

mlie788 committed:
Removed unused parameters and imports

mlie788 committed:
Changed min width of login form

mlie788 committed:
Changed min width of login form

tcodling reviewed (commented):
From a code point of view this is awesome! Just that little bit of state weirdness I mentioned with the "Forgot password" screen.

Just waiting on response from megan/felicity around those missing screens

mlie788 committed:
Fixed issue with states and added the navigation to reset password back in and styled

mlie788 committed:
Fixed issue with states and added the navigation to reset password back in and styled

passcod committed:
Deduplicate gitignores

passcod committed:
Remove old electron package.json

passcod committed:
Revert servers back to commonjs

mlie788 committed:
Styling reset password workflow

mlie788 committed:
Styling reset password workflow

passcod committed:
Host vite on localhost

passcod committed:
Add clean:deps script

passcod committed:
Dual-build shared packages

passcod committed:
Now that desktop build is gone, we can specify node 20 only

passcod committed:
Use es6 style in constants' source

passcod committed:
Change shared's export

passcod committed:
Write app version via Vite

passcod committed:
Merge branch 'main' into feature/tav-676-browser-edition

passcod committed:
Add dummy codeship so it stops complaining

passcod committed:
what the heck

passcod committed:
codeship does nothing (please)

mlie788 committed:
Restored previous order of reset password screens

mlie788 committed:
Restored previous order of reset password screens

mlie788 committed:
Cleaned up frogot password states (only clear when press forgot password, added error messages back in)

mlie788 committed:
Cleaned up frogot password states (only clear when press forgot password, added error messages back in)

mlie788 committed:
Fixed error message functionality

mlie788 committed:
Fixed error message functionality

mlie788 committed:
Restyling change password workflow to latest design (without functionality))

mlie788 committed:
Restyling change password workflow to latest design (without functionality))

mlie788 committed:
Added top margin to password success text

mlie788 committed:
Added top margin to password success text

passcod committed:
Use alls-green action

passcod committed:
More jsx from merge

mlie788 committed:
Refactored showing error text

mlie788 committed:
Refactored showing error text

mlie788 committed:
Moved inline styling for success page

mlie788 committed:
Moved inline styling for success page

mlie788 committed:
Changed version number to 2.0

mlie788 committed:
Merge branch 'feature/tav-676-browser-edition' into feature/nass-988-tamanu-desktop-login-page-update

mlie788 committed:
Merge branch 'feature/tav-676-browser-edition' into feature/nass-988-tamanu-desktop-login-page-update

mlie788 committed:
Merge branch 'feature/nass-988-tamanu-desktop-login-page-update' of github.com:beyondessential/tamanu into feature/nass-988-tamanu-desktop-login-page-update

invalid-email-address committed:
Bump version to 2.1.0

passcod committed:
Merge release/2.0 to main (#5111)

* Cut-off for release/2.0

* Fix web: remove mentions of process.env (#5107)

* Fix: shim process for misbehaving libraries (#5108)

* Allow NODE_ENV

* Contain global to one util

---------

Co-authored-by: github-actions <github-actions@github.com>

flynncs committed:
NASS-1072: MultiPageHeader component for pdfs (#5094)

* implemented header

* moved header to shared

* lint

tcaiger committed:
yarn lint (#5115)

* yarn lint

* tweaks

flynncs committed:
NASS-1075: DataSection component for pdfs (#5076)

* implemented data section

* removed redundant imports & testing

* more redundant imports

* unused prop

* moved to shared

* linting

* moved again

* render multiple data items

function for rendering multiple data items with localisation and accessors

* lint & react loop key

---------

Co-authored-by: Tom Caiger <caigertom@gmail.com>

flynncs committed:
NASS-1070: Footer component for pdfs (#5077)

* implemented footer

* separators

* updated path

* moved to shared

* reloacted footer

* lint (auto fix)

* Revert "lint (auto fix)"

This reverts commit 7d13b93c0c232af8f663b490584dcbd9718cd51d.

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Tom Caiger <caigertom@gmail.com>

mcccclean committed:
Lint fix (#5118)

flynncs committed:
implemented patient details component (#5116)

Co-authored-by: Tom Caiger <caigertom@gmail.com>

flynncs committed:
NASS-1073: PatientDetailsWithAddress component for PDF's (#5121)

* implemented component & accessor

* changed accessor & multi rendering

mcccclean committed:
Fix watch-shared-and build script (#5123)

flynncs committed:
NASS-1069: PatientDetailsWithBarcode component for PDF's (#5122)

* implemented component - work req.

need to update the barcode display to work in react pdf

* barcode using jsbarcode

* accessor change & barcode styling

* lint

* Update PrintableBarcode.jsx

---------

Co-authored-by: Tom Caiger <caigertom@gmail.com>

tcaiger committed:
Merge branch 'release/2.0' into merge/2-0-to-main

tcaiger committed:
Merge pull request #5127 from beyondessential/merge/2-0-to-main

Merge/2 0 to main

mlie788 committed:
Merge branch 'feature/nass-988-tamanu-desktop-login-page-update' of github.com:beyondessential/tamanu into feature/nass-988-tamanu-desktop-login-page-update

mlie788 committed:
Merge branch 'main' into feature/nass-988-tamanu-desktop-login-page-update

mlie788 committed:
Revert "Merge branch 'main' into feature/nass-988-tamanu-desktop-login-page-update"

This reverts commit bf23808cf4382f16edead5250fc0ace1df06dfc7, reversing
changes made to dc19910f50f84c2a702e56aea2b43562d0aef11c.

mlie788 committed:
Updated yarn.lock file

BhargavDY committed:
EPI-726: tooltip date issue resolved (#5104)

passcod committed:
Add hint about needing to remove the rewrite: line (#5132)

mlie788 committed:
Merge branch 'main' into feature/nass-988-tamanu-desktop-login-page-update

mlie788 committed:
Revert "Merge branch 'main' into feature/nass-988-tamanu-desktop-login-page-update"

This reverts commit 573ac873c582ad2b37edd0380a52927b32d1ba73, reversing
changes made to 1d14caf9932a10c96427dc7cdbf9f44525bc4c11.
